<a href="https://colab.research.google.com/github/1804054Miraz/My_Thesis_Work/blob/main/Count_csv_files_in_the_directories.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>



Total alcoholic data in new_alcoholic directory is :  3849

Total alcoholic data in Transfer_new_non_alcoholic directory is :  3811



In [ ]:
import os
import tarfile
import gzip
import pandas as pd
import glob
import numpy as np


In [ ]:
import os
total_count=0
path = "/content/drive/MyDrive/Dataset/EEG_FULL/Anomaly/Alcoholic/"
csv_file_count = {}
i=0
# Initialize a dictionary to store the count of CSV files for each directory
anomaly_alcoholic = {}

# Iterate through each subdirectory
for directory_name in os.listdir(path):
    directory_path = os.path.join(path, directory_name)

    # Check if the path is a directory
    if os.path.isdir(directory_path):
        # Count the number of CSV files in the directory
        csv_files = [file for file in os.listdir(directory_path) if file.endswith('.csv')]
        csv_count = len(csv_files)

        # Store the count in the dictionary
        anomaly_alcoholic[str(i)+"_"+directory_name] = csv_count
        csv_file_count[str(i)+"_"+directory_name] = csv_count
        i+=1

# Print the count for each directory
total = 0
for directory, count in anomaly_alcoholic.items():
    print(f"Directory {directory}: {count} CSV files")
    total += count
print("Total alcoholic data in anomaly directory is : ", total)
total_count += total

FileNotFoundError: ignored

In [ ]:
path = "/content/drive/MyDrive/Dataset/EEG_FULL/Non_selected_alcoholic/"

# Initialize a dictionary to store the count of CSV files for each directory
nonselected_alcoholic = {}

# Iterate through each subdirectory
for directory_name in os.listdir(path):
    directory_path = os.path.join(path, directory_name)

    # Check if the path is a directory
    if os.path.isdir(directory_path):
        # Count the number of CSV files in the directory
        csv_files = [file for file in os.listdir(directory_path) if file.endswith('.csv')]
        csv_count = len(csv_files)

        # Store the count in the dictionary
        nonselected_alcoholic[directory_name] = csv_count
        csv_file_count[str(i)+"_"+directory_name] = csv_count
        i+=1

# Print the count for each directory
total = 0
for directory, count in nonselected_alcoholic.items():
    # print(f"Directory {directory}: {count} CSV files")
    total += count
print("Total Non-selected-alcoholic data in EEG_FULL directory is : ", total)
total_count += total

Total Non-selected-alcoholic data in EEG_FULL directory is :  2133


In [ ]:


path = "/content/drive/MyDrive/Dataset/EEG_FULL/Anomaly/Non-alcoholic/"

# Initialize a dictionary to store the count of CSV files for each directory
anomaly_non_alcoholic = {}

# Iterate through each subdirectory
for directory_name in os.listdir(path):
    directory_path = os.path.join(path, directory_name)

    # Check if the path is a directory
    if os.path.isdir(directory_path):
        # Count the number of CSV files in the directory
        csv_files = [file for file in os.listdir(directory_path) if file.endswith('.csv')]
        csv_count = len(csv_files)

        # Store the count in the dictionary
        anomaly_non_alcoholic[directory_name] = csv_count
        csv_file_count[str(i)+"_"+directory_name] = csv_count
        i+=1

# Print the count for each directory
total = 0
for directory, count in anomaly_non_alcoholic.items():
    # print(f"Directory {directory}: {count} CSV files")
    total += count
print("Total non-alcoholic data in anomaly directory is : ", total)
total_count += total

Total non-alcoholic data in anomaly directory is :  111


In [ ]:


path = "/content/drive/MyDrive/Dataset/EEG_FULL/Old data/Alcoholic"

# Initialize a dictionary to store the count of CSV files for each directory
anomaly_non_alcoholic = {}
csv_file_count = {}
i=0
# Iterate through each subdirectory
for directory_name in os.listdir(path):
    directory_path = os.path.join(path, directory_name)

    # Check if the path is a directory
    if os.path.isdir(directory_path):
        # Count the number of CSV files in the directory
        csv_files = [file for file in os.listdir(directory_path) if file.endswith('.csv')]
        csv_count = len(csv_files)

        # Store the count in the dictionary
        anomaly_non_alcoholic[directory_name] = csv_count
        csv_file_count[str(i)+"_"+directory_name] = csv_count
        i+=1

# Print the count for each directory
total = 0
for directory, count in anomaly_non_alcoholic.items():
    # print(f"Directory {directory}: {count} CSV files")
    total += count
print("Total alcoholic data in EEG_FULL directory is : ", total)
# total_count += total

Total alcoholic data in EEG_FULL directory is :  3849


In [ ]:
import os

path = "/content/drive/MyDrive/Dataset/EEG_FULL/Old data/Non-alcoholic/"

# Initialize a dictionary to store the count of CSV files for each directory
anomaly_non_alcoholic = {}
# Iterate through each subdirectory
for directory_name in os.listdir(path):
    directory_path = os.path.join(path, directory_name)

    # Check if the path is a directory
    if os.path.isdir(directory_path):
        # Count the number of CSV files in the directory
        csv_files = [file for file in os.listdir(directory_path) if file.endswith('.csv')]
        csv_count = len(csv_files)

        # Store the count in the dictionary
        anomaly_non_alcoholic[directory_name] = csv_count
        csv_file_count[str(i)+"_"+directory_name] = csv_count
        i+=1

# Print the count for each directory
total = 0
for directory, count in anomaly_non_alcoholic.items():
    print(f"Directory {directory}: {count} CSV files")
    total += count
print("Total Non-alcoholic data in EEG_FULL directory is : ", total)
total_count += total

In [ ]:
sorted_csv_file_count = dict(sorted(csv_file_count.items(), key=lambda item: item[1], reverse=True))
print("Total count of EEG_FULL is : ", len(sorted_csv_file_count))
print(len(sorted_csv_file_count))
total = 0
for directory, count in sorted_csv_file_count.items():
    # print(f"Directory {directory}: {count} CSV files")
    total += count
print("Total Data in EEG_FULL directory is : ", total)
total_count += total

Total count of EEG_FULL is :  122
122
Total Data in EEG_FULL directory is :  10955


In [ ]:

# Set the path to the main directory
path = "/content/drive/MyDrive/Dataset/eeg_full/"

# # Initialize a dictionary to store the count of rd.gz files for each tar.gz file
# rd_count_by_tar = {}

# # Iterate through each tar.gz file
# for tar_file_name in os.listdir(path):
#     tar_file_path = os.path.join(path, tar_file_name)

#     # Check if the path is a file and ends with .tar.gz
#     if os.path.isfile(tar_file_path) and tar_file_name.endswith('.tar.gz'):
#         # Extract the contents of the tar.gz file to a temporary directory
#         with tarfile.open(tar_file_path, 'r:gz') as tar:
#            # List all the files in the archive
#           file_list = tar.getnames()
#           # print(file_list)
#           # Initialize a list to store DataFrames
#           dfs = []

#           # Loop through the files in the archive
#           for file_name in file_list:
#               if file_name.endswith('.gz'):
#                   extracted_file = tar.extractfile(file_name)
#                   decompressed_data = gzip.GzipFile(fileobj=extracted_file)
#                   df = pd.read_csv(decompressed_data, delimiter='\t')
#                   dfs.append(df)
#           rd_count_by_tar[file_name]=len(dfs)
#           # print(file_name, "has ",len(dfs))



In [ ]:
# sorted(rd_count_by_tar)
# # Print the count for each tar.gz file
# total = 0;
# sorted_rd_count_by_tar = dict(sorted(rd_count_by_tar.items(), key=lambda item: item[1], reverse=True))
# print(len(rd_count_by_tar))
# for tar_file, count in sorted_rd_count_by_tar.items():
#     # print(f"Tar file {tar_file}: {count} .gz files")
#     total += count

print("Total alcoholic data in eeg_full directory is : ", total)

122
Total alcoholic data in eeg_full directory is :  11074


In [ ]:
import os
total_count=0
path = "/content/drive/MyDrive/Dataset/EEG_FULL/New_alcoholic/"
csv_file_count = {}
i=0
# Initialize a dictionary to store the count of CSV files for each directory
anomaly_alcoholic = {}

# Iterate through each subdirectory
for directory_name in os.listdir(path):
    directory_path = os.path.join(path, directory_name)

    # Check if the path is a directory
    if os.path.isdir(directory_path):
        # Count the number of CSV files in the directory
        csv_files = [file for file in os.listdir(directory_path) if file.endswith('.csv')]
        csv_count = len(csv_files)

        # Store the count in the dictionary
        anomaly_alcoholic[str(i)+"_"+directory_name] = csv_count
        csv_file_count[str(i)+"_"+directory_name] = csv_count
        i+=1

# Print the count for each directory
total = 0
for directory, count in anomaly_alcoholic.items():
    # print(f"Directory {directory}: {count} CSV files")
    total += count
print("Total alcoholic data in new_alcoholic directory is : ", total)
total_count += total

Total alcoholic data in new_alcoholic directory is :  3811


In [ ]:
## Transpose the dataframe. Now the dataframe are 64x256. We want 256x64 with columns name also
new_path ='/content/drive/MyDrive/Dataset/EEG_FULL/Transfer_new_non_alcoholic/'
# output_directory = "/content/drive/MyDrive/Dataset/EEG_FULL/New_non_alcoholic/"
# non_alcoholic_path = '/content/drive/MyDrive/Dataset/EEG_FULL/Non-alcoholic/'

t=0
# print(alcoholic_path)
# ! mkdir new_path+directory_name
for directory_name in os.listdir(new_path):
  # !mkdir {new_path + directory_name}
  # time.sleep(5)
  directory_path = os.path.join(new_path, directory_name)
  # print(directory_path)
  if os.path.isdir(directory_path):
    csv_files = [file for file in os.listdir(directory_path) if file.endswith('.csv')]
    count = len(csv_files)
    t+=count
    # print(count)
    # for file in csv_files:
    #   # print(directory_path+'/'+file)
    #   df = pd.read_csv(directory_path+'/'+file)
    #   df = df.transpose()
    #   df.columns=['FP1', 'FP2', 'F7', 'F8', 'AF1', 'AF2', 'FZ', 'F4', 'F3', 'FC6', 'FC5', 'FC2', 'FC1', 'T8', 'T7', 'CZ', 'C3', 'C4', 'CP5', 'CP6', 'CP1', 'CP2', 'P3', 'P4', 'PZ', 'P8', 'P7', 'PO2', 'PO1', 'O2', 'O1', 'X', 'AF7', 'AF8', 'F5', 'F6', 'FT7', 'FT8', 'FPZ', 'FC4', 'FC3', 'C6', 'C5', 'F2', 'F1', 'TP8', 'TP7', 'AFZ', 'CP3', 'CP4', 'P5', 'P6', 'C1', 'C2', 'PO7', 'PO8', 'FCZ', 'POZ', 'OZ', 'P2', 'P1', 'CPZ', 'nd', 'Y']
    #   df.to_csv(new_path+directory_name+'/'+file, index=False)
print("Total Non-alcoholic data in alcoholic directory is : ", t)

Total Non-alcoholic data in alcoholic directory is :  3811


Total Non-alcoholic data in EEG_FULL directory is :  3811

Total alcoholic data in EEG_FULL directory is :  3849

Total Data in EEG_FULL directory is :  10955

Total alcoholic data in eeg_full directory is :  11074